### DISCUSSION TITLE
Treatlife Dimmer switch showing up as switch only

### DISCUSSION THREAD

**Discussion by nteuscher** [milo526/homebridge-tuya-web#501] [Q&A] [answered]
I have Treatlife WiFi Dimmer switches in my Tuya app that work properly ([link to product page](https://www.treatlife.tech/collections/smart-dimmer-switch/products/smart-dimmer-switch-treatlife-wifi-light-switch-for-dimmable-led-halogen-incandescent-bulbs-compatible-with-alexa-google-assistant-ifttt-remote-control-single-pole-neutral-wire-required-4-pack?variant=31727505834045)). When adding them to Homebridge, they show up as simple on/off switches with no ability to adjust the intensity. I don't see an option for "dimmer switch" as a Device Type in the custom setup. 

So when these appear in HomeKit, they show up as simple on/off switches only with no ability to control intensity. Are there any options for me to make these dimmable in HomeKit?

**Comment by milo526** [comment, marked as answer]
Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)

Without this it is hard for me to help you with regards to this specific issue.

**Comment by nteuscher** [comment]
Thanks ... I followed those instructions and those devices came back as type "light". I set that in the device settings and now they work properly :) 
Thanks!
